modules/digitaluniverse/rendering/renderablepoints.cpp

```diff
@@ -91,118 +81,99 @@ namespace {
     };
+
+    struct [[codegen::Dictionary(RenderablePoints)]] Parameters {
+        // The path to the SPECK file that contains information about the astronomical
+        // object being rendered
+        std::string file;
+
+        // Astronomical Object Color (r,g,b)
+        glm::dvec3 color;
+
+        enum class Unit {
+            Meter [[codegen::key("m")]],
+            Kilometer [[codegen::key("Km")]],
+            Parsec [[codegen::key("pc")]],
+            Kiloparsec [[codegen::key("Kpc")]],
+            Megaparsec [[codegen::key("Mpc")]],
+            Gigaparsec [[codegen::key("Gpc")]],
+            Gigalightyears [[codegen::key("Gly")]]
+        };
+        std::optional<Unit> unit;
+
+        // [[codegen::verbatim(SpriteTextureInfo.description)]]
+        std::optional<std::string> texture;
+
+        // [[codegen::verbatim(ScaleFactorInfo.description)]]
+        std::optional<float> scaleFactor;
+
+        // [[codegen::verbatim(ColorMapInfo.description)]]
+        std::optional<std::string> colorMap;
+    };
+#include "renderablepoints_codegen.cpp"
 }  // namespace
 
 namespace openspace {
 
 documentation::Documentation RenderablePoints::Documentation() {
-    using namespace documentation;
-    return {
-        "RenderablePoints",
-        "digitaluniverse_renderablepoints",
-        {
-            {
-                "Type",
-                new StringEqualVerifier("RenderablePoints"),
-                Optional::No
-            },
-            {
-                KeyFile,
-                new StringVerifier,
-                Optional::No,
-                "The path to the SPECK file that contains information about the "
-                "astronomical object being rendered."
-            },
-            {
-                keyColor,
-                new Vector3Verifier<double>,
-                Optional::No,
-                "Astronomical Object Color (r,g,b)."
-            },
-            {
-                SpriteTextureInfo.identifier,
-                new StringVerifier,
-                Optional::Yes,
-                SpriteTextureInfo.description
-            },
-            {
-                ScaleFactorInfo.identifier,
-                new DoubleVerifier,
-                Optional::Yes,
-                ScaleFactorInfo.description
-            },
-            {
-                ColorMapInfo.identifier,
-                new StringVerifier,
-                Optional::Yes,
-                ColorMapInfo.description
-            },
-
-        }
-    };
+    documentation::Documentation doc = codegen::doc<Parameters>();
+    doc.id = "digitaluniverse_renderablepoints";
+    return doc;
 }
 
-
 RenderablePoints::RenderablePoints(const ghoul::Dictionary& dictionary)
     : Renderable(dictionary)
     , _scaleFactor(ScaleFactorInfo, 1.f, 0.f, 64.f)
     , _pointColor(
         ColorInfo,
         glm::vec3(1.f, 0.4f, 0.2f),
         glm::vec3(0.f, 0.f, 0.f),
-        glm::vec3(1.0f, 1.0f, 1.0f)
+        glm::vec3(1.f, 1.f, 1.f)
```

Comment:
`glm::vec3(1.f)`? and `glm::vec3(0.f)` above?

Reply:
? I don't understand?   Unless I'm blind, I just replaced `1.0f` with `1.f`?

Reply:
This was just another super small refactoring suggestion while poking around in these classes anyways, to write the color as: 
```
, _pointColor(ColorInfo, glm::vec3(1.f, 0.4f, 0.2f), glm::vec3(0.f), glm::vec3(1.f))
```

Would make it one line and I also think it makes it clearer that the color ranges from black to white. But feel free to ignore it 😄